src/main/java/org/opensearch/knn/index/MethodComponentContext.java

```diff
@@ -193,6 +197,69 @@ public Map<String, Object> getParameters() {
         return parameters;
     }
 
+    @Override
+    public String toString() {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("{name=").append(name).append(";");
+        stringBuilder.append("parameters=[");
+        if (Objects.nonNull(parameters)) {
+            for (Map.Entry<String, Object> entry : parameters.entrySet()) {
+                stringBuilder.append(entry.getKey()).append("=");
+                String value = entry.getValue().toString();
+                value = value.replace(",", "$%$");
+                stringBuilder.append(value).append(";");
+            }
+        }
+        stringBuilder.append("]}");
+        return stringBuilder.toString();
+    }
+
+    public static MethodComponentContext fromString(String in) {
+        int index = 0;
+
+        String[] outerMethodComponentContextArray = in.split("\\{", -1);
```

Comment:
Why do we need "\\"?

Reply:
The split method takes a regex as an input.  { is a special character in regex so the \\ escapes it to just split by {

Reply:
Can we avoid this by just confirming the first and last character are {} and then building a substring from then which is processed.

Reply:
The split still needs to be there for the recursion to work.  Otherwise, I can't pass the whole substring of a nested MethodComponentContext because it will be split by ;